woocommerce/woocommerce
# Add error handling and tests for ShippingPartnerSuggestions

### Changes proposed in this Pull Request:

Partially closes #44248.



### How to test the changes in this Pull Request:

#### Test shipping task

1. In a fresh store, go through onboarding, select `United Kingdom` country in onboarding wizard
2. In homescreen, click on `Add shipping costs` task
3. Fill in the form with whatever and continue in step 1 and 2
4. In step 3, observe `ShipStation` and `sendcloud` is shown

#### Testing transient when specs are valid

1. Open console with access to `wp-cli`
2. Run `wp transient get timeout_woocommerce_admin_shipping_partner_suggestions_specs`
3. Confirm it returns a timestamp that is about 7 days from now 

#### Testing transient when specs are invalid

2. Manually replace [ShippingPartnerSuggestionsDataSourcePoller.php](https://github.com/woocommerce/woocommerce/blob/7440082381f633d76a7fe2cddb3f026877cdb888/plugins/woocommerce/src/Admin/Features/ShippingPartnerSuggestions/ShippingPartnerSuggestionsDataSourcePoller.php#L21) with https://api.npoint.io/dedddcb56c3642d967f9
3. Run `wp transient delete woocommerce_admin_shipping_partner_suggestions_specs`
4. Run `wp transient delete timeout_woocommerce_admin_shipping_partner_suggestions_specs`
5. Run `wp option delete woocommerce_admin_created_default_shipping_zones`
6. Run `wp option delete woocommerce_admin_reviewed_default_shipping_zones`
7. Go to `WooCommerce > Settings > Shipping` and delete all shipping zones.
2. Go to homescreen, click on `Add shipping costs` task
3. Run `wp transient get timeout_woocommerce_admin_shipping_partner_suggestions_specs`
7. Confirm it returns a timestamp that is about 3 hours from now 

### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
